aicq/blue.py

```
@@ -0,0 +1,383 @@
+            self.aic = aiclib.nvp.Connection(uri)
+        return self.aic
+
+    @property
+    def connection_description(self):
+        return self._get_connection
+
+    def _get_connection(self):
+        if not self.conn_error:
+            return self.conn
+        self.aic = None
+        min_errors = sys.maxint
+        ret = self.conn
+        for conn in self.connections:
+            if conn['errors'] < min_errors:
+                ret = conn
+                min_errors = conn['errors']
+        self.conn = ret
+        return self.conn
+
+    @property
+    def default_zone(self):
+        return self._get_connection['default_tz']
+
+    def _connection_error(self, connection):
+        self.conn_error = True
+        connection['errors'] += 1
+
+    def connection_test(self):
+        return self.connection.nvp_function().logout()
+
+# --------------------------------
+# NVP utility functions
+# --------------------------------
+
+    def default_transport_zone_exists(self):
+        """This will check if the default transport zone for the current
+        connection actually exists"""
+        try:
+            self.connection.zone(self.default_zone).read()
+        except aiclib.nvp.ResourceNotFound:
+            return False
+        return True
+
+    def check_tenant(self, net_id, tenant_id):
+        """Returns true of the tenant 'owns' this network"""
+        network = self.get_network(net_id)
+        for t in network["tags"]:
+            if t["scope"] == "os_tid" and t["tag"] == tenant_id:
+                return True
+        return False
+
+# --------------------------------
+# Network (lswitch) functions
+# --------------------------------
+
+    def get_network(self, net_id):
+        resp = self.connection.lswitch(net_id).read()
+        return resp
+
+    def check_network_existance(self, net_id):
+        try:
+            self.get_network(net_id)
+            return True
+        except Exception:
+            pass
```

I would at least LOG.debug the actual exception here. While I agree with the assessment here that an exception indicates a missing network, I'm afraid it would accidentally mask other exceptions masquerading as this.